### PR TITLE
[Fix](Nereids)add nereids load function in read fields of GlobalFunctionMgr and Database

### PR DESCRIPTION
**Pull request by sohardforaname**
## Proposed changes

Issue Number: close #xxx

cherry-pick from master #23248
commit-id: c3327b51b9eb69b6189e527e50dc55fb123a1a15

add nereids load function in read fields of GlobalFunctionMgr and Database to fix some udf is lost when restart FE and
query with Nereids.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by sohardforaname** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 44.54 seconds
 stream load tsv:          547 seconds loaded 74807831229 Bytes, about 130 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17164544632 Bytes

File: fe/fe-core/src/main/java/org/apache/doris/catalog/FunctionUtil.java [hunk 1]
```diff
@@ -234,8 +239,11 @@ public static boolean translateToNereids(String dbName, Function function) {
                 JavaUdaf.translateToNereidsFunction(dbName, ((AggregateFunction) function));
             }
         } catch (Exception e) {
-            LOG.warn("Nereids create function {}:{} failed", dbName, function.getFunctionName().getFunction(), e);
+            LOG.warn("Nereids create function {}:{} failed, caused by: {}", dbName == null ? "_global_" : dbName,
+                    function.getFunctionName().getFunction(), e);
         }
+        LOG.info(String.format("Nereids add function: %s:%s OK!", dbName == null ? "_global_" : dbName,
+                function.getFunctionName().getFunction()));
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/FunctionUtil.java:246]: remove this log

File: fe/fe-core/src/main/java/org/apache/doris/catalog/FunctionUtil.java [hunk 2]
```diff
@@ -246,8 +254,11 @@ public static boolean dropFromNereids(String dbName, FunctionSearchDesc function
                     .collect(Collectors.toList());
             Env.getCurrentEnv().getFunctionRegistry().dropUdf(dbName, fnName, argTypes);
         } catch (Exception e) {
-            LOG.warn("Nereids drop function {}:{} failed", dbName, function.getName(), e);
+            LOG.warn("Nereids drop function {}:{} failed, caused by: {}", dbName == null ? "_global_" : dbName,
+                    function.getName(), e);
         }
+        LOG.info(String.format("Nereids drop function: %s:%s OK!", dbName == null ? "_global_" : dbName,
+                function.getName()));
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/FunctionUtil.java:261]: remove this log

**Review by morrySnow — CHANGES_REQUESTED** [review]

**Comment by sohardforaname** [comment]
run buildall

**Comment by sohardforaname** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.49 seconds
 stream load tsv:          547 seconds loaded 74807831229 Bytes, about 130 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.7 seconds inserted 10000000 Rows, about 336K ops/s
 storage size: 17162220219 Bytes

**Review by morrySnow — APPROVED** [review]

**Comment by sohardforaname** [comment]
run p0

**Pull request merged**